### ISSUE TITLE
Feature request: Git-review integration

### ISSUE THREAD

**Issue by jamesharris-garmin** [Gruntfuggly/gerrit-view#8]
It would be extremely nice if this tool could be integrated with the git-review tool:

https://docs.openstack.org/infra/git-review/

My thoughts here are:
we could use the .gitreview file to source information like workspace or open/file gerrit urls.

we could also use this to let us select a patch and then quickly pull the latest change set for review.  Does this seem like something that would be interesting for your company?

**Comment by Gruntfuggly** [comment]
I would love to extend the extension to provide more features for reviewing code. At some point I would like to at least show review comments inline with the code and allow comments to be added.

I had a brief look at git-review, and it seems like it could be fairly easily implemented, although it might take a bit of work to make it work cross-platform as it looks like it is completely command line based.

Unfortunately, I have too much other stuff going on at the moment - writing vscode plugins is basically a spare time hobby. 

It might be simplest to write a separate git-review extension for vscode first and then integrate with gerrit-view later?

**Comment by jamesharris-garmin** [comment]
That might be a better approach for integrating git-review commands or "check out a review branch". though Git-review is a python based tool so it does work on windows as well.

I was more thinking that you could automatically limit your search based on the project information in the git review file (that format isn't exactly hard to parse.)

**Comment by Gruntfuggly** [comment]
Can you post an example git review file?

**Comment by jamesharris-garmin** [comment]
https://docs.openstack.org/infra/git-review/installation.html#gitreview-file-format is the documentation from the openstack project.

**Comment by Gruntfuggly** [comment]
Ah OK - I'm with you now - It's a way to adjust the query automatically.  I'll see what I can do.

**Comment by jamesharris-garmin** [comment]
Technically I would also like this to be a point of configuration per-project as to which gerrit server we are querying. (i.e. if I am working on an openstack project I would like to reference openstack's gerrit server. etc.)

This is related to #9

**Comment by Gruntfuggly** [comment]
How about if I read the .git-review file (if present) and if it contains  `gerrit-view=false` then I'll disable it for the workspace?

I presume that the contents of any .git-review file should overwrite the standard vscode settings?

**Comment by jamesharris-garmin** [comment]
That wasn't necessarily what I meant. When I mentioned #9. I am more interested in not being pestered to add my gerrit-server IP address when I am not in a project that uses gerrit or the gerrit view. 

I would have to test that adding additional settings to .gitreview doesn't break the `git-review` tool.

**Comment by Gruntfuggly** [comment]
OK, so for #9, just have a `gerrit-view.enabled` setting which you can set to `false` in the user (global) settings and then set it to `true` per workspace that needs it?

**Comment by jamesharris-garmin** [comment]
Maybe it would be better to phrase the behavior I would like this way:

1. gerrit-view should not prompt to automatically globally input a gerrit-server. If one is not configured, the view should not show up. If one is configured then the view should be initialized/populated.
2. gerrit-view should autodetect the server from a '.gitreview' file if present and only populate the server at the workspace configuration level. (at the user level). 
3. gerrit-view should, by default limit queries of the server to the project mentioned in the `.gitreview` file if the file is detected.
4. The user should be able to disable `.gitreview` detection and fall back to the internal settings for queries and server location.

I think this is the general behavior I want. (The biggest thing here is to not prompt me to enter a server name if I haven't configured one) Maybe on first load of the plugin a notification could be sent to prompt the user to configure a server and link them to the appropriate setting in the configuration.

**Comment by Gruntfuggly** [comment]
Would it be workable (as long as I make it clear in the prompt) that if no server is entered then gerrit-view is disabled for the workspace (and remembered)? I would rather do that than have some sort of first load flag.

**Comment by jamesharris-garmin** [comment]
As long as it didn't prompt me if I wasn't already in a folder that had a git repo. The view probably shouldn't exist unless the scm plugin is activated for the workspace.  I just don't want its default behavior in a new window to be "hey! what is your gerrit server! Let me fetch EVERYTHING from it!". That can make for really slow startups.

I know that other plugins (like git-lens) hook the "I just installed" event to show notifications that we updated and we should do something. Maybe you could prompt then?
